wasm/wasm_source/src/tx_bond.rs

```diff
@@ -108,41 +107,86 @@ mod tests {
             .read(&pos_balance_key)?
             .expect("PoS must have balance");
         assert_eq!(pos_balance_pre, initial_stake);
-        let total_voting_powers_pre = ctx().read_total_voting_power()?;
+
+        // Read some data before the tx is executed
+        let total_deltas_pre = ctx().read_total_deltas()?;
+        let validator_deltas_pre =
+            ctx().read_validator_deltas(&bond.validator)?.unwrap();
         let validator_sets_pre = ctx().read_validator_set()?;
-        let validator_voting_powers_pre =
-            ctx().read_validator_voting_power(&bond.validator)?.unwrap();
 
         apply_tx(ctx(), tx_data)?;
 
-        // Read the data after the tx is executed
+        // Read the data after the tx is executed.
+        let validator_deltas_post =
+            ctx().read_validator_deltas(&bond.validator)?.unwrap();
+        let total_deltas_post = ctx().read_total_deltas()?;
+        let validator_sets_post = ctx().read_validator_set()?;
 
         // The following storage keys should be updated:
 
-        //     - `#{PoS}/validator/#{validator}/total_deltas`
-        let total_delta_post =
-            ctx().read_validator_total_deltas(&bond.validator)?;
-        for epoch in 0..pos_params.pipeline_len {
-            assert_eq!(
-                total_delta_post.as_ref().unwrap().get(epoch),
-                Some(initial_stake.into()),
-                "The total deltas before the pipeline offset must not change \
-                 - checking in epoch: {epoch}"
-            );
-        }
-        for epoch in pos_params.pipeline_len..=pos_params.unbonding_len {
-            let expected_stake =
-                i128::from(initial_stake) + i128::from(bond.amount);
-            assert_eq!(
-                total_delta_post.as_ref().unwrap().get(epoch),
-                Some(expected_stake),
-                "The total deltas at and after the pipeline offset epoch must \
-                 be incremented by the bonded amount - checking in epoch: \
-                 {epoch}"
-            );
+        //     - `#{PoS}/validator/#{validator}/deltas`
+        //     - `#{PoS}/total_deltas`
+        //     - `#{PoS}/validator_set`
+
+        // Check that the validator set and deltas are unchanged before pipeline
+        // length and that they are updated between the pipeline and
+        // unbonding lengths
+        //
+        // TODO: should end be pipeline + unbonding now?
```

Comment:
not in here, I think, in tx_unbond once we make the change (let's do that when we switch to LazyMap based epoched)